### ISSUE TITLE
Distance configuration improvement (for better understandability)

### ISSUE THREAD

**Issue by GuillaumeDelaporte** [open-io/oio-sds#728]
We moved from a configuration like `distance=1` for the first level of location, `distance=2` for the second level, and so on ... to a mask to select the right level of location with `distance=1` every time.

It makes the configuration trickier to understand.

**Comment by jfsmig** [comment]
**TLDR** more config options and more automation are necessary IMO

Well well well ...

It seems the original goal of the load-balancing rework is still pretty far. At least is it not obvious. So we'll remind it.

We had a problem too complex to be solved efficiently (i.e. in a predictable amount of time). We had to poll sets of items with inter-distance constraints, and common tags constraints. This gave us a kind of a [SAT problem](https://en.wikipedia.org/wiki/Boolean_satisfiability_problem). We decided a simplification was necessary.

We chose to build **bags** of services matching all the criteria for a load-balancing request. And because the requests are known in advance, it is possible to persist the bags. E.g. a bag contains all the `rawx` services tagged as `fast`.

Before, it was only possible to specify to poll *N* services with a given inter-distance applied to each pair of services. Now it is possible to specify you want *N⁰* services in a given zone, and *N¹* others in an other zone. This is possible because the distance constraints have been removed from the bag definition to allow fine-tuned settings. So the case you describe (and regret) where `distance>1` is still possible, you *just* have to organise the services into clever bags.

With bags we have a solution to long-distance constraints, but we do not manage yet the close-proximity problem between services chosen into the same bag. Since a bag manages services with a `distance<=1` constraint between pairs of services. In other words, `distance=1` or `distance=0`, respectively services poll'able in the same request or not.

Then we introduced the weak notion of **bucket** to figure services proximity. Services in a same bucket are too close, services in different bucket are far enough.

I am personally convinced this way is efficient and flexible, as soon as we give the proper configuration keys. And this is lacking.

Now It is only possible to configure a single value for the location of a service. I consider the configuration could be much easier with 2 values:
- 1 for the bag, let's call it the **zone**
- 1 for the bucket, let's call it the **position**

We also must provide dynamic descriptions for bags, instead of explicitly specifying all the bags each service belongs to. E.g.: `tag.zone==france.marseille && tag.class==fast && type==rawx`

@racciari, @GuillaumeDelaporte what do you think about this ?

**Comment by jfsmig** [comment]
In an obvious absence of reply, I consider the point is "not so clear" for the moment.

There is no planned change to come back to a configuration shaped with "distance=x" like directives (this would lead to the return of NP-complete constraints).

Let's move to a bit of documentation.
